### PR TITLE
Add map to search

### PR DESCRIPTION
**Pull request by IonatanSala**
When user searches using the search bar it will redirect him to a search page. All the shop locations in that search area will be displayed on the map. The products in them shops are also listed on the page.

### CODE REVIEW

**Comment by danielpaul** [comment]
@IonatanSala make the map be on the right hand side of the search results. Check airbnb for example. or this for example: https://www.justpark.com/uk/parking/dublin/?filter=1&q=dublin&ac_country=&place_id=&location_name=&source=none&coords=&start_date=31+Mar+2016&end_date=31+Mar+2016

Also the marker content needs to be designed like the attached screenshot in trello. 
- [ ]  Add image, address and name.
- [ ]  Need custom marker as icons.
- [ ] Call to action button in the popup to visit shop page. 

image

**Comment by IonatanSala** [comment]
@danielpaul I'm just going to add the other search into this branch

**Comment by danielpaul** [comment]
@IonatanSala okey cool.

File: app/views/search/index.html.haml [hunk 1]
```diff
@@ -35,19 +36,31 @@
...
+          <img class="img-fluid" src="' + (locations[i]["header_image"]["url"] || "http://placehold.it/250x250") + '" > \
+        </div> \
+      </div> ';
+
     marker = new google.maps.Marker({
       position: new google.maps.LatLng(locations[i]["latitude"], locations[i]["longitude"]),
       map: map,
-      maxWidth: 200
+      maxWidth: 200,
+      icon: "http://www.megaicons.net/static/img/icons_sizes/22/119/32/location-map-icon.png"
```
Comment (danielpaul) [inline app/views/search/index.html.haml]: Need to serve this from our assets

File: app/controllers/search_controller.rb [hunk 2]
```diff
@@ -1,11 +1,35 @@
 class SearchController < ApplicationController
   def index
-    if params[:search]
+
+    if params[:search] && (!params[:search][:lat].blank? || !params[:search][:lng].blank?)
```
Comment (danielpaul) [inline app/controllers/search_controller.rb:4]: Could refactor and use a builder pattern but we can refactor later.

File: app/views/pages/home.html.haml [hunk 3]
```diff
@@ -1,7 +1,39 @@
 .container
   .col-md-8.col-md-offset-2.text-xs-center
-    %h1 Lorem Ipsum is simply dummy text.
-    %h5 Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
+    -# %form.form-inline
+    -#   .form-group
```
Comment (danielpaul) [inline app/views/pages/home.html.haml]: A lot of comment outs. Could we remove the code that we are not using?

Comment (danielpaul) [inline app/views/pages/home.html.haml]: Codebase needs to be clean with just the code we need. If we need this commented out code later, we can get it back from git history.

File: app/views/search/index.html.haml (hunk 1 above)
Comment (IonatanSala) [inline app/views/search/index.html.haml]: Okay.

**Pull request merged**